packages/ast-spec/src/unions/ObjectLiteralElement.ts

```diff
@@ -1,8 +1,7 @@
-import type { MethodDefinition } from '../element/MethodDefinition/spec';
 import type { Property } from '../element/Property/spec';
 import type { SpreadElement } from '../element/SpreadElement/spec';
 
-export type ObjectLiteralElement = MethodDefinition | Property | SpreadElement;
+export type ObjectLiteralElement = Property | SpreadElement;
```

Comment:
Huh, I guess they're not the same in objects. Cool.

Reply:
Yeah objects use `Property` with `method: true`

There's definitely scope to work on the `Property` type to better discriminate the union based on the flags.
I might raise a PR to do that soon as it's a pretty nice property to have hyper-specific types for this sort of case.

I actually did it when I wrote the flow-estree types
https://github.com/facebook/hermes/blob/main/tools/hermes-parser/js/hermes-estree/src/types.js#L448-L536
(looks like I didn't do method types, but that's a natural next step)